### PR TITLE
Add "why µsort" page to docs, update readme

### LINKED ISSUE #116: What is the difference between isort and this repo
Are there any advantages to use this repo over [isort](https://github.com/PyCQA/isort)?

### PR DESCRIPTION
**Pull request by jreese**
Adds new page to docs site, "Why µsort":

- [x] Overview of safety, use at scale, and how
- [x] Design goals
- [x] Non-goals
- [x] Comparison to isort 

Preview docs: https://usort--130.org.readthedocs.build/en/130/

Closes #116

### CODE REVIEW

File: docs/why.rst [hunk 1]
```diff
@@ -0,0 +1,111 @@
+Why µsort?
+==========
+
+µsort was originally designed with safety as the top priority–"first, do no harm".
```
Comment (thatch) [inline docs/why.rst:4]: I would prefer an em dash with spaces either side.  As-rendered this is an en dash?

File: docs/why.rst [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+- `LibCST`_ for parsing files and modifying syntax trees.
+  µsort transforms CST nodes into dataclasses, then back to CST nodes after sorting.
+  This guarantees grammatical correctness of both the input file and the generated
+  output. Support for future grammar or syntax changes is the responsibility of LibCST.
+
+- No support for configuring output style. µsort works best when run before a
+  dedicated code formatter like `Black`_ for enforcing style choices. µsort does use
+  the configured line length for Black when rendering imports on one or more lines.
+
+- No vendored code. All dependencies are satisfied from PyPI, with liberal version
```
Comment (thatch) [inline docs/why.rst:49]: "liberal version constraints" doesn't have a well-defined meaning for me.  Maybe mention that the version ranges are as wide as possible, future-proof assuming SemVer works, and validated (to some extent) by CI?

File: docs/why.rst [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+- Removing unused imports. This would require µsort to also consume and understand the
+  functionality of each file to accurately know which imports are unused, which would
+  increase the amount of code and testing needed to guarantee safety. Furthemore, the
+  possibility for import-time side effects in removed imports makes this operation
+  potentially dangerous in subtle ways.
+
+- Configurable output style. More options means more code paths to follow, and more
+  test cases needed to cover all possible outputs. This is better left for dedicated
+  code formatters like `Black`_.
```
Comment (thatch) [inline docs/why.rst:70]: I think this is a good spot to link ufmt.

File: docs/why.rst [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+Comparison to isort
+-------------------
+
+Unlike `isort`_, µsort only sorts imports within individual "blocks" of imports,
+inferred based on a number of heuristics (see :ref:`Import Blocks <import-blocks>`
+for details). µsort will not attempt to move imports outside of these blocks, or
+across any blocks of code that aren't import statements.
+
+For example, µsort detects two distinct blocks of imports in the following code::
```
Comment (thatch) [inline docs/why.rst:84]: Might also be worth calling out the human intent of something more like the isort readme's example

```
import a
print("after a")
import b
```

...as people commonly do in debug w/ prints.

File: docs/why.rst (hunk 1 above)
Comment (jreese) [reply docs/why.rst:4]: It was meant to be an em dash, but I frequently miss pressing shift in time and get an en dash instead. 😅  I will fix that.

That said, most style guides recommend *not* having spaces around the em dash. Eg, https://www.merriam-webster.com/words-at-play/em-dash-en-dash-how-to-use and https://style.mla.org/is-it-wrong-to-type-spaces-before-and-after-a-dash/

**Pull request merged**